### PR TITLE
Adding projects folder change test

### PR DESCRIPTION
**Pull request by iterion**
Closes https://github.com/KittyCAD/modeling-app/issues/3369

This is my first playwright test for modeling-app so please nitpick! Just copy-pasta from other tests and some digging on the weird hoop for handling file chooser. I added a new testid as well as it made it much easier to select the right thing in the DOM.

The rest of the changes are necessary to support Nixos. I'm happy to tweak them if desired but I do need to be able to point to specific executables and not the stuff playwright & electron both download.

### CODE REVIEW

File: e2e/playwright/projects.spec.ts [hunk 1]
```diff
@@ -605,3 +605,114 @@ test(
...
+            `${dir}/bracket/main.kcl`
+          ),
+        ])
+      },
+    })
+    await page.setViewportSize({ width: 1200, height: 500 })
+
+    page.on('console', console.log)
+
+    await page.waitForTimeout(1_000)
```
Comment (franknoirot) [inline e2e/playwright/projects.spec.ts:636]: I don't think you should need this timeout: was it helping you somehow?

File: e2e/playwright/projects.spec.ts [hunk 2]
```diff
@@ -605,3 +605,114 @@ test(
...
+      await fsp.rm(newProjectDirName, { recursive: true })
+    }
+
+    await test.step('We can change the root project directory', async () => {
+      // expect to see the project directory settings link
+      await expect(
+        page.getByTestId('project-directory-settings-link')
+      ).toBeVisible()
+
+      await page.waitForTimeout(100)
```
Comment (franknoirot) [inline e2e/playwright/projects.spec.ts:653]: Similarly don't think you need this here, the `click()` function from Playwright has what they call "[auto waiting](https://playwright.dev/docs/actionability)", setting a little timeout to check for actionability

File: e2e/playwright/projects.spec.ts [hunk 3]
```diff
@@ -605,3 +605,114 @@ test(
...
+        [newProjectDirName]
+      )
+      page.getByTestId('project-directory-button').click()
+
+      await expect(page.locator('section#projectDirectory input')).toHaveValue(
+        newProjectDirName
+      )
+
+      page.getByTestId('settings-close-button').click()
+      await page.waitForTimeout(100)
```
Comment (franknoirot) [inline e2e/playwright/projects.spec.ts:677]: Should be able to remove

**Review by franknoirot — APPROVED** [review]
This is so good! I just asked if it's possible to remove a few timeouts, Playwright auto waiting should cover them if that's what you have them in there for. But if it breaks the tests to remove don't worry about it.

File: e2e/playwright/projects.spec.ts [hunk 4]
```diff
@@ -605,3 +605,114 @@ test(
...
+
+    await test.step('We can change the root project directory', async () => {
+      // expect to see the project directory settings link
+      await expect(
+        page.getByTestId('project-directory-settings-link')
+      ).toBeVisible()
+
+      await page.waitForTimeout(100)
+
+      page.getByTestId('project-directory-settings-link').click()
```
Comment (jtran) [inline e2e/playwright/projects.spec.ts:655]: I think all these yellow squiggles need to be awaited.

Run `yarn eslint e2e`. There's probably already some warnings on `main`, but I'm hoping we can avoid adding new issues.

Screenshot 2024-08-16 at 11 02 48 AM

Comment (franknoirot) [reply e2e/playwright/projects.spec.ts:655]: Great catch! Yeah not awaiting clicks can definitely lead to racy behavior

Comment (iterion) [reply e2e/playwright/projects.spec.ts]: Thanks, this explains my races!

**Pull request merged**